g2p_registry_rest_api/services/group.py

```diff
@@ -0,0 +1,312 @@
+import logging
+
+from odoo.addons.base_rest import restapi
+from odoo.addons.base_rest_pydantic.restapi import (
+    PydanticModel,
+    PydanticModelList,
+)
+from odoo.addons.component.core import Component
+
+from ..models.group import (
+    GroupInfoIn,
+    GroupInfoOut,
+    GroupShortInfoOut,
+)
+from ..models.group_search_param import GroupSearchParam
+
+
+class GroupApiService(Component):
+    _inherit = "base.rest.service"
+    _name = "registrant_group.rest.service"
+    _usage = "group"
+    _collection = "base.rest.registry.services"
+    _description = """
+        Registrant Group API Services
+    """
+
+    @restapi.method(
+        [
+            (
+                [
+                    "/<int:id>",
+                ],
+                "GET",
+            )
+        ],
+        output_param=PydanticModel(GroupInfoOut),
+        auth="user",
+    )
+    def get(self, _id):
+        """
+        Get partner's information
+        """
+        partner = self._get(_id)
+        return GroupInfoOut.from_orm(partner)
+
+    @restapi.method(
+        [(["/", "/search"], "GET")],
+        input_param=PydanticModel(GroupSearchParam),
+        output_param=PydanticModelList(GroupShortInfoOut),
+        auth="user",
+    )
+    def search(self, partner_search_param):
+        """
+        Search for partners
+        :param partner_search_param: An instance of partner.search.param
+        :return: List of partner.short.info
+        """
+        domain = []
+        if partner_search_param.name:
+            domain.append(("name", "like", partner_search_param.name))
+        if partner_search_param.id:
+            domain.append(("id", "=", partner_search_param.id))
+        if partner_search_param.is_group:
+            domain.append(("is_group", "=", True))
```

Comment:
How do you search for not group? Remove this param. is_group should be set by default in this API.

Reply:
Fixed